### PR TITLE
#166647233 enable authors to delete their own articles

### PR DESCRIPTION
**Pull request by Ramadhan0**
##### Have this point working 
* DELETE ```/api/articles/:slug```: this is the endpoint, Author will hit whenever he/she wants to delete his/her own articles
#### How should this be manually tested?
* Clone the repo by typing  ```git clone https://github.com/andela/savage_rangers-ah``` into the terminal

* Install development dependencies by type ```npm install```
* Run the server with this command ```npm run dev```
* Test it with your postman typing:
 
   ```localhost:3001/api/articles/:slug```

#### What are the relevant pivotal tracker stories?
[166647233](https://www.pivotaltracker.com/story/show/166647233)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,6 +81,8 @@
       "src/configs/environments.js",
       "src/configs/redisConfigs.js",
       "src/configs/passport.js",
+      "src/index.js",
```
Comment (MCFrank16) [inline package.json:89]: why do you have 2 index files in here, Kindly remove one

File: src/api/models/index.js [hunk 2]
```diff
@@ -8,6 +8,8 @@ const sequelize = new Sequelize(env.dbUrl, {
 
 const models = {
   User: sequelize.import('./user'),
+  Article: sequelize.import('./article'),
+  SequelizeMeta: sequelize.import('./sequelizemeta')
```
Comment (MCFrank16) [inline src/api/models/index.js]: we don't need the sequelizemeta file anymore, please remove it

File: src/api/models/sequelizemeta.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export default (sequelize, DataTypes) => {
```
Comment (MCFrank16) [inline src/api/models/sequelizemeta.js]: remove this file

File: src/api/models/test-articl.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+'use strict';
```
Comment (MCFrank16) [inline src/api/models/test-articl.js]: please change this file to ES6, and do you really need this? if yes for what?

File: src/api/seeders/20190624215911-Article.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-unused-vars */
```
Comment (MCFrank16) [inline src/api/seeders/20190624215911-Article.js]: no eslint disable

**Review by MCFrank16 — CHANGES_REQUESTED** [review]

**Comment by MCFrank16** [comment]
change your PR naming convention and follow the engineer playbook

File: src/api/controllers/ArticleController.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+   *
+   * @static
+   * @param {Object} req - request object
+   * @param {Object} res - respond object
+   * @returns {Object} response body
+   * @memberof deleteArticle
+   */
+  static async delete(req, res) {
+    const { slug } = req.params;
+    await Article.destroy({ where: { slug } });
```
Comment (truestbyheart) [inline src/api/controllers/ArticleController.js:23]: what if it fails what happens?

File: tests/deleteArticle.test.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  });
+
+  it('should delete article', (done) => {
+    chai
+      .request(server)
+      .delete('/api/articles/How-to-create-sequalize-seeds')
+      .set('Authorization', `${userToken}`)
+      .end((err, res) => {
+        res.should.have.status(status.OK);
+        res.body.should.have.status(status.OK);
```
Comment (truestbyheart) [inline tests/deleteArticle.test.js]: why check the status of the response body and the response at the same time

File: tests/deleteArticle.test.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  it('should fail to delete article', (done) => {
+    chai
+      .request(server)
+      .delete('/api/articles/This-is-it')
+      .set('Authorization', `${userToken}`)
+      .end((err, res) => {
+        res.should.have.status(status.ACCESS_DENIED);
+        res.body.should.have
+          .property('message')
+          .eql('Please you must be the owner of this Article in order to modify it, Thanks');
```
Comment (truestbyheart) [inline tests/deleteArticle.test.js:49]: Put this message int the error messages object

File: src/middlewares/tokenVerifier.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import 'dotenv/config';
+import jwt from 'jsonwebtoken';
+import statuses from '../helpers/constants/status.codes';
```
Comment (truestbyheart) [inline src/middlewares/tokenVerifier.js]: why don't you use  status instead of statuses?

File: src/middlewares/tokenVerifier.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  const token = req.headers.authorization;
+
+  try {
+    const payloadData = await jwt.verify(token, env.secret);
+    req.user = payloadData;
+    next();
+  } catch (err) {
+    res.status(statuses.UNAUTHORIZED).json({
+      status: statuses.UNAUTHORIZED,
+      message: 'Forbiden access'
```
Comment (truestbyheart) [inline src/middlewares/tokenVerifier.js]: why don't you put this . in the error messages object and use the error sender to send the error?

File: src/middlewares/checkArticleOwnership.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+   * @memberof CheckUserOwnership
+   * @returns {Object} res
+   */
+  static async checkOwner(req, res, next) {
+    const {
+      user: { id }
+    } = req.user;
+    const { slug } = req.params;
+
+    const response = await Article.findOne({
```
Comment (truestbyheart) [inline src/middlewares/checkArticleOwnership.js:29]: same here, why don't you handle both success and failure operations?

File: src/middlewares/checkArticleOwnership.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+        userId: id
+      }
+    });
+
+    if (response) {
+      req.Existing = response.dataValues;
+      next();
+    } else {
+      res.status(statusCode.ACCESS_DENIED).json({
+        message: 'Please you must be the owner of this Article in order to modify it, Thanks'
```
Comment (truestbyheart) [inline src/middlewares/checkArticleOwnership.js:41]: find a short and meaningful response for this action like ``You don't have permission to delete this article``

File: README.md [hunk 13]
```diff
@@ -420,6 +420,11 @@ No additional parameters required
 
 `git clone https://github.com/andela/savage_rangers-ah.git`
 
+- Install dependecies by runing
+- Clone the repo using:
+
+`git clone https://github.com/andela/savage_rangers-ah.git`
```
Comment (truestbyheart) [inline README.md:425]: Git clone to your local machine is enough.

**Review by truestbyheart — CHANGES_REQUESTED** [review]

**Comment by MemunaHaruna** [comment]
@Ramadhan0 do you see anything wrong with your branch name? `ft-deleteArticle-#166647233`?

File: .eslintignore [hunk 14]
```diff
@@ -16,6 +16,7 @@ coverage
 
 # Migrations
 src/api/migrations
+src/api/models/test-articl.js
```
Comment (MemunaHaruna) [inline .eslintignore]: @Ramadhan0 sigh

File: README.md [hunk 15]
```diff
@@ -420,6 +420,11 @@ No additional parameters required
 
 `git clone https://github.com/andela/savage_rangers-ah.git`
 
+- Install dependecies by runing
```
Comment (MemunaHaruna) [inline README.md]: @Ramadhan0 rebase properly with develop. This file shouldn't be here.

File: src/middlewares/checkArticleOwnership.js (hunk 11 above)
Comment (MemunaHaruna) [reply src/middlewares/checkArticleOwnership.js:29]: @Ramadhan0 please attend to this feedback

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**